src/test/ui/self/arbitrary_self_types_pin_lifetime_impl_trait.stderr

```diff
@@ -1,13 +1,13 @@
-error: cannot infer an appropriate lifetime
+error[E0758]: cannot infer an appropriate lifetime
   --> $DIR/arbitrary_self_types_pin_lifetime_impl_trait.rs:6:44
    |
 LL |     fn f(self: Pin<&Self>) -> impl Clone { self }
    |                ----------     ----------   ^^^^ ...and is captured here
    |                |              |
-   |                |              ...is required to be `'static` by this...
-   |                data with this lifetime...
+   |                |              ...is required to live as long as `'static` by this...
+   |                this data with an anonymous lifetime `'_`...
    |
-help: to permit non-static references in an `impl Trait` value, you can add an explicit bound for the anonymous lifetime #1 defined on the method body at 6:5
+help: to permit non-static references in an `impl Trait` value, you can add an explicit bound for an anonymous lifetime `'_`
```

Comment:
I feel like it'd be good if we could take this opportunity to explain lifetime elision a bit better. How hard would it be to say

help: to declare that you capture borrowed data from `self`, you can add an explicit `'_` bound

Reply:
Current output:

<img width="853" alt="Screen Shot 2020-06-02 at 4 06 58 PM" src="https://user-images.githubusercontent.com/1606434/83578244-1ed88e80-a4eb-11ea-8aee-7d109786715c.png">